city_scrapers/spiders/il_pollution_control.py

```diff
@@ -0,0 +1,164 @@
+import json
+from datetime import datetime
+from urllib.parse import urljoin
+
+import scrapy
+from city_scrapers_core.constants import BOARD, CANCELLED, FORUM, NOT_CLASSIFIED, PASSED, TENTATIVE
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class IlPollutionControlSpider(CityScrapersSpider):
+    name = "il_pollution_control"
+    agency = "Illinois Pollution Control Board"
+    timezone = "America/Chicago"
+    allowed_domains = ["pcb.illinois.gov"]
+    start_urls = ["https://pcb.illinois.gov/ClerksOffice/MeetingMinutes"]
+    json_url = "https://pcb.illinois.gov/ClerksOffice/GetCalendarEvents"
+
+    def __init__(self, *args, **kwargs):
+        self.link_map = dict()  # Populated by self._parse_minutes()
+        self.link_parse_complete = False
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def from_crawler(cls, crawler, *args, **kwargs):
+        """ Overridden `from_crawler` to connect `spider_idle` signal. """
+        spider = super(IlPollutionControlSpider, cls).from_crawler(crawler, *args, **kwargs)
+        crawler.signals.connect(spider.spider_idle, signal=scrapy.signals.spider_idle)
+        return spider
+
+    def spider_idle(self):
+        """ React to `spider_idle` signal by starting JSON parsing after _parse_minutes."""
+        if not self.link_parse_complete:
```

Comment:
First off, the `spider_idle` setup is really well done! We're using that in some of the pipelines in `city-scrapers-core`, but this is a really solid application.

I don't think you need the `link_parse_complete` attribute here though. Because you're disconnecting from the `spider_idle` handle in the method it can only run once which should achieve the same think as a boolean flag for checking whether it's done.

Reply:
Thanks for the kind words! I bet you're right, I'll look into it. That boolean was in there with a handful of other stuff when I was in the "how on earth do I make this work" phase of development, and I never even considered it might've become redundant once I got it working.